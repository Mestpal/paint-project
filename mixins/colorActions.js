import { mapActions, mapGetters } from 'vuex'

export default {
  computed: {
    ...mapGetters('colorManager', [
      'selectedColor',
      'swatcherColors'
    ])
  },
  methods: {
    ...mapActions('colorManager', [
      'updateSelectedColor'
    ]),
    updateColorAction (colorEvent) {
      this.updateSelectedColor(colorEvent)
    }
  }
}
