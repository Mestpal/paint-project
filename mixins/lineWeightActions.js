import { mapActions, mapGetters } from 'vuex'

export default {
  computed: {
    ...mapGetters('LineWeightManager', [
      'selectedLineWeight',
      'lineWeightsOptions'
    ])
  },
  methods: {
    ...mapActions('LineWeightManager', [
      'updateLineSelectedWeight'
    ]),
    updateLineWeightAction (lineEvent) {
      this.updateLineSelectedWeight(lineEvent)
    }
  }
}
