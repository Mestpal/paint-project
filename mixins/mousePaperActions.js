import paper from 'paper'
import { mapActions } from 'vuex'

export default {
  methods: {
    ...mapActions('mouseDrawing', [
      'updateDrawing'
    ]),
    updateDrawingAction (diameter) {
      this.updateDrawing({
        paper,
        diameter
      })
      // paper.setup(document.getElementById('paint_project_canvan'))
      // // Now, draw your things based on component state.
      // const point = new paper.Point(20, 20)
      // const circle = new paper.Path.Circle(point, this.circleDiameter / 2)
      // circle.fillColor = 'red'
      // circle.strokeColor = 'blue'
    }
  }
}
