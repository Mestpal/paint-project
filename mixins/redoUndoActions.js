import { mapActions } from 'vuex'

export default {
  computed: {
    // ...mapGetters('colorManager', [
    //   'selectedColor',
    //   'swatcherColors'
    // ])
  },
  methods: {
    ...mapActions('drawHistoryManager', [
      'undoLastDrawStep',
      'redoLastDrawStep',
      'saveDrawStepsHistory'
    ]),
    undoDrawStepAction () {
      this.undoLastDrawStep()
    },
    redoDrawStepAction () {
      this.redoLastDrawStep()
      console.log('redoDrawStepAction')
    },
    saveDrawHistoy () {
      this.saveDrawStepsHistory()
    }
  }
}
