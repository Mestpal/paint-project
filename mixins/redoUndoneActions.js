import { mapActions } from 'vuex'

export default {
  computed: {
    // ...mapGetters('colorManager', [
    //   'selectedColor',
    //   'swatcherColors'
    // ])
  },
  methods: {
    ...mapActions('drawHistoryManager', [
      'undoLastDrawStep'
    ]),
    undoDrawStepAction () {
      console.log('undoDrawStepAction')
      this.undoLastDrawStep()
    },
    redoDrawStepAction () {
      console.log('redoDrawStepAction')
    }
  }
}
