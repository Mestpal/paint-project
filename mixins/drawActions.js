import { mapActions, mapGetters } from 'vuex'
import colorActions from '@/mixins/drawModules/colorActions'
import redoUndoActions from '@/mixins/drawModules/redoUndoActions'
import lineWeightActions from '@/mixins/drawModules/lineWeightActions'

export default {
  mixins: [
    colorActions,
    redoUndoActions,
    lineWeightActions
  ],
  computed: {
    ...mapGetters('drawingStatus', [
      'drawStatus'
    ])
  },
  methods: {
    ...mapActions('drawingStatus', [
      'updateDrawStatus'
    ]),
    ...mapActions('paperjsStore', [
      'setCanvasProject',
      'initPath',
      'updatePath'
    ]),
    createDrawPathAction (newPointEvent) {
      newPointEvent.selectedColor = this.selectedColor
      newPointEvent.selectedLineWeight = this.selectedLineWeight
      this.initPath(newPointEvent)
      this.updateDrawStatus(true)
    },
    updateDrawPathAction (newPointEvent) {
      if (this.drawStatus) {
        this.updatePath(newPointEvent)
      }
    },
    closeDrawPathAction (newPointEvent) {
      this.updatePath(newPointEvent)
      this.updateDrawStatus()
      this.saveDrawHistoy()
    }
  },
  mounted () {
    this.setCanvasProject()
  }
}
