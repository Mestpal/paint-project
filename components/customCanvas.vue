<template>
  <v-col :cols="8">
    <v-card>
      <v-card-text>
        <canvas
          id="paint_project_canvan"
          @mousedown="createDrawPathAction($event)"
          @mousemove="updateDrawPathAction($event)"
          @mouseup="closeDrawPathAction($event)"
          style = "height: 600px; width: 100%; background-color: white"
        >
          {{ circleDiameter }}
        </canvas>
      </v-card-text>
    </v-card>
  </v-col>
</template>

<script>
import drawActions from '../mixins/drawActions'

export default {
  mixins: [drawActions],
  props: {
    'circleDiameter': {
      type: Number,
      default: 10
    }
  },
  data: () => ({ x: 20, y: 20 }),
  methods: {
    test (text) {
      console.log(text)
    }
  }
}
</script>
