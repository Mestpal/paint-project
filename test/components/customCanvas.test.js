
import Vue from 'vue'
import Vuetify from 'vuetify'
import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils'

import customCanvas from '@/components/customCanvas'
import { getters } from '@/mixins/drawActions'

Vue.use(Vuetify)
Vue.use(Vuex)
const localVue = createLocalVue()

describe('customCanvas.vue', () => {
  let store
  let wrapper

  beforeEach(() => {
    store = new Vuex.Store({
      getters
    })
    wrapper = shallowMount(customCanvas, {
      store,
      localVue
    })
  })

  it('customCanvas component renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
