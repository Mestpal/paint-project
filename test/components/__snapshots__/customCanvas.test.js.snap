// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`customCanvas.vue customCanvas component renders correctly 1`] = `
<v-col-stub cols="8" tag="div">
  <v-card-stub loaderheight="4" tag="div">
    <v-card-title-stub> Canvas </v-card-title-stub>
    <v-card-text-stub><canvas id="paint_project_canvan" style="height: 600px; width: 100%; background-color: white;"></canvas></v-card-text-stub>
  </v-card-stub>
</v-col-stub>
`;
